Assets/MXR.SDK/Runtime/Android/MXRAndroidSystem.cs

```diff
@@ -439,7 +448,16 @@ public class MXRAndroidSystem : IMXRSystem {
             }
         }
 
-        void InitializeDeviceStatus() {
+        void TryInitializeDeviceStatus() {
+            // On Android30 and above, we check if we have storage access. Otherwise file read attempt will fail.
+            if (MXRAndroidUtils.AndroidSDKAsInt >= 30 && !MXRAndroidUtils.IsExternalStorageManager) {
+                if (LoggingEnabled)
+                    Debug.unityLogger.LogWarning(TAG, "Not initializing DeviceStatus using local json, "
+                    + "the SDK will fallback to trying to initialize it using the Android Messenger. This is not an error. "
+                    + EXTERNAL_STORAGE_MANAGER_WARNING_MSG);
+                return;
+            }
```

Comment:
Can we make this a function and use it in both places this is called?

Reply:
+ 1